microprofile/cdi/src/main/java/io/helidon/microprofile/cdi/BuildTimeInitializer.java

```diff
@@ -34,21 +39,38 @@ final class BuildTimeInitializer {
     private BuildTimeInitializer() {
     }
 
-    static synchronized HelidonContainerImpl get() {
-        if (null == container) {
-            createContainer();
-        }
+    static HelidonContainerImpl get() {
+        return accessContainer(() -> {
+            if (null == container) {
+                createContainer();
+            }
 
-        return container;
+            return container;
+        });
     }
 
-    static synchronized void reset() {
-        container = null;
+    static void reset() {
+        accessContainer(() -> {
+            container = null;
+            return null;
+        });
     }
 
     private static void createContainer() {
         // static initialization to support GraalVM native image
-        container = HelidonContainerImpl.create();
-        ContainerInstanceHolder.addListener(BuildTimeInitializer::reset);
+        accessContainer(() -> {
+            container = HelidonContainerImpl.create();
+            ContainerInstanceHolder.addListener(BuildTimeInitializer::reset);
+            return null;
+        });
+    }
+
+    private static <T> T accessContainer(Supplier<T> operation) {
+        CONTAINER_ACCESS.writeLock().lock();
```

Comment:
I haven't seen a call to `readLock()` yet, which makes me think that using `ReentrantLock` instead of `ReentrantReadWriteLock` would be a better choice.

Reply:
Correct. Fixed.